### ISSUE TITLE
pgcli doesn't reconnect on SSL SYSCALL error: EOF detected

### ISSUE THREAD

**Issue by relistan** [dbcli/pgcli#643]
The ability to reconnect was added in #210  but it doesn't seem to reconnect when `SSL SYSCALL error: EOF detected` is received.

```
foo@mydb:mydb> SELECT count(1) FROM account;
SSL SYSCALL error: EOF detected

foo@mydb:mydb> SELECT count(1) FROM account;
connection already closed
```

**Comment by amjith** [comment]
@relistan Good to hear from you. 😄 

Sorry about the error. Would you be able to give me a repro case for this? I can't trigger a `SSL SYSCALL error`.

**Comment by amjith** [comment]
I'm able to semi-reproduce it. 

1. I launched pgcli and then killed postgres. 
2. Tried a command in pgcli.
3. Pgcli asks me if I'd like to reconnect.
4. I say 'Yes'. But reconnect fails since postgres is still not running.
5. Now I launch postgres and try a different command in pgcli.
6. Pgcli returns `connection already closed`. 

Is this what you see? Or do you not get a prompt to reconnect when you get the first error?

BTW, what version of pgcli are you running `pgcli --version`? 

The latest version is 1.4.0, if you installed it via brew I'd recommend upgrading `brew update && brew upgrade pgcli`

**Comment by relistan** [comment]
Hey man! Good to hear from you, too 😃 . It seems to be when the network connection goes away under it. Yes, what you describe is what I see except it never asks to reconnect. I was on 1.3.1 which is the latest that `brew update` can find. But I removed it with brew and was able to `pip install` 1.4.0.

Using 1.4.0 I still don't get a request to reconnect. What I see is:
```
foo@mydb:mydb> SELECT "count"(1) FROM account
SSL SYSCALL error: EOF detected

foo@mydb:mydb>

Time: 0.000s
foo@mydb:mydb> SELECT "count"(1) FROM account
connection already closed
foo@mydb:mydb> SELECT "count"(1) FROM account
connection already closed```

**Comment by relistan** [comment]
Looks like on Linux I can reproduce by connecting, inserting an IPtables rule to block outbound packets, running a query, then removing the IPtables rule:

```
<in another terminal, connect pgcli>
kmatthias@ubuntu:~$ sudo iptables -I OUTPUT 1 --proto tcp --dport 5432 -j DROP
<in another terminal run a query, wait for SSL SYSCALL error>
kmatthias@ubuntu:~$ sudo iptables -D OUTPUT 1```

**Comment by amjith** [comment]
Thanks Karl. I'll take a stab at this tonight. I appreciate the commands to repro this case. :)

**Comment by amjith** [comment]
I have fixed this in a PR #644. Once that gets merged we'll start a new release.

**Comment by relistan** [comment]
Awesome, @amjith! That was quick!

**Comment by j-bennet** [comment]
Fixed in 1.5.0.

**Comment by relistan** [comment]
Thanks @amjith @j-bennet !
